hooks/emitter/feeds.go

```diff
@@ -0,0 +1,117 @@
+package emitter
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/bandprotocol/chain/v3/hooks/common"
+	"github.com/bandprotocol/chain/v3/x/feeds/types"
+)
+
+func (h *Hook) emitSetFeedsSignalTotalPower(stp types.Signal) {
+	h.Write("SET_SIGNAL_TOTAL_POWER", common.JsDict{
+		"signal_id": stp.ID,
+		"power":     stp.Power,
+	})
+}
+
+func (h *Hook) emitRemoveFeedsSignalTotalPower(signalID string) {
+	h.Write("REMOVE_SIGNAL_TOTAL_POWER", common.JsDict{
+		"signal_id": signalID,
+	})
+}
+
+func (h *Hook) emitSetFeedsVoterSignal(ctx sdk.Context, voter string, signal types.Signal) {
+	h.Write("SET_VOTER_SIGNAL", common.JsDict{
+		"voter":     voter,
+		"signal_id": signal.ID,
+		"power":     signal.Power,
+		"timestamp": ctx.BlockTime().UnixNano(),
+	})
+}
+
+func (h *Hook) emitRemoveFeedsVoterSignals(voter string) {
+	h.Write("REMOVE_VOTER_SIGNALS", common.JsDict{
+		"voter": voter,
+	})
+}
+
+func (h *Hook) emitSetFeedsSignalPricesTx(ctx sdk.Context, txHash []byte, validator string, feeder string) {
+	h.Write("SET_SIGNAL_PRICES_TX", common.JsDict{
+		"tx_hash":   txHash,
+		"validator": validator,
+		"feeder":    feeder,
+		"timestamp": ctx.BlockTime().UnixNano(),
+	})
+}
+
+func (h *Hook) emitSetFeedsValidatorPrices(ctx sdk.Context, validator string, signalPrices []types.SignalPrice) {
+	h.Write("SET_VALIDATOR_PRICES", common.JsDict{
+		"validator":     validator,
+		"signal_prices": signalPrices,
+		"timestamp":     ctx.BlockTime().UnixNano(),
+	})
+}
+
+func (h *Hook) emitSetFeedsPrices(ctx sdk.Context, prices []types.Price) {
+	h.Write("SET_PRICES", common.JsDict{
+		"prices":    prices,
+		"timestamp": ctx.BlockTime().UnixNano(),
+	})
+}
+
+func (h *Hook) emitSetFeedsReferenceSourceConfig(ctx sdk.Context, rsc types.ReferenceSourceConfig) {
+	h.Write("SET_REFERENCE_SOURCE_CONFIG", common.JsDict{
+		"registry_ipfs_hash": rsc.RegistryIPFSHash,
+		"registry_version":   rsc.RegistryVersion,
+		"timestamp":          ctx.BlockTime().UnixNano(),
+	})
+}
+
+// handleMsgVote implements emitter handler for MsgVote.
+func (h *Hook) handleFeedsMsgVote(
```

Comment:
Ensure method names and comments are aligned for clarity.

Reply:
ok